enaml/core/parser.py

```diff
@@ -3445,6 +3445,56 @@ def p_arglist14(p):
     p[0] = Arguments(keywords=keywords, starargs=p[2], kwargs=p[8])
 
 
+def _validate_arglist_and_kwlist(p, items, keywords):
+    kwnames = set()
+    args = []
+    kws = []
+    _validate_arglist_list(items, p.lexer.lexer)
+    for arg in items:
+        if isinstance(arg, ast.keyword):
+            kws.append(arg)
+            kwnames.add(arg.arg)
+        else:
+            args.append(arg)
+    for kw in keywords:
+        if not isinstance(kw, ast.keyword):
+            msg = 'only named arguments may follow *expression'
+            tok = FakeToken(p.lexer.lexer, p.lineno(1))
```

Comment:
It would for now, but could lead to a bug in the future if we refactor. I *think* that's why inlined the keyword checks originally.